docs/WindowsBuild.md

```diff
@@ -0,0 +1,176 @@
+# Building Swift on Windows
+
+This document describes how to build Swift for Windows natively. See [the
+Windows doc](./Windows.md) for more about what is possible with Swift on
+Windows.
+
+There are two supported ways to build Swift on Windows, they are
+
+1. Using [`clang-cl`](https://clang.llvm.org/docs/UsersManual.html#clang-cl)
+1. Using the Microsoft Visual C++ compiler (MSVC)
+
+`clang-cl` is recommended over MSVC for building Swift on Windows.
+Although it is possible to build the compiler and the standard library with
+MSVC, and use those built products to compile a Swift program, it won't be
+possible to run the binary without seperately obtaining the Swift runtime. On
+the other hand, `clang-cl` is able to build the runtime, which makes it
+possible to build and run all the components required for Swift natively on
+Windows.
+
+## MSVC
+- Windows doesn't currently have a build script. You'll need to run commands
+  manually to build Swift on Windows.
+- Windows support for Swift is a work in progress and may not work on your
+  system, but it has been tested.
+- Using the latest Visual Studio version is recommended. Swift may fail to
+  build with older C++ compilers.
+
+### 1. Install dependencies
+- Make sure to add Python, CMake and Ninja to your `Path` environment variable
```

Comment:
Maybe move this below the list of dependencies to go along with the other "Make sure to..."?